src/Behat/Context/EntityContext.php

```diff
@@ -355,11 +355,27 @@ public function entityCreate($entity_type, $entity) {
   /**
    * @AfterScenario
    */
-  public function deleteGeneratedEntities() {
-    foreach ($this->entities as $entity_type => $ids) {
-      $this->getCore()->entityDeleteMultiple($entity_type, $ids);
+  public function cleanEntities() {
+
+    $bypass_entities = isset($this->customParameters['entities_clean_bypass']) ? $this->customParameters['entities_clean_bypass'] : FALSE;
+
+    if ($bypass_entities) {
+      $entities = array_filter($this->entities, function($key) use ($bypass_entities){
+        return !in_array($key, $bypass_entities);
+      }, ARRAY_FILTER_USE_KEY);
+    }
+    else {
+      $entities = $this->entities;
     }
-  }
 
+    // Array reversion is needed in order to guarantee that any hierarchical
+    // relationship between entities (i.e: group and subgroup) is respected
+    // by removing first the later (dependent) entities.
+    foreach (array_reverse($entities) as $entity_type => $ids) {
+      foreach (array_reverse($ids) as $id) {
+        $this->getCore()->entityDeleteMultiple($entity_type, [$id]);
```

Comment:
```suggestion
        $this->getCore()->entityDelete($entity_type, $id);
```

Reply:
The reason is due to Groups module. Groups create related entities (group content) that can't be removed before the group. If the group content is removed before the group, an error is thrown. This bypass can mitigate this problem.

Reply:
This is another comment, I'll reply in the correct comment: https://github.com/Metadrop/behat-contexts/pull/100/files#r627490006